hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/FilterParams.java

```diff
@@ -17,45 +22,61 @@
  * Encapsulates a set of filters that may be specified (optionally). Properties
  * that are not specified (e.g. <code>null</code> for simple properties) When
  * applied, these filters are AND-gated.
- *
  */
-public class FilterParams {
+public class FilterParams implements Serializable {
+    private static final long serialVersionUID = 1L;
 
     private final Collection<TargetUpdateStatus> filterByStatus;
     private final Boolean overdueState;
     private final String filterBySearchText;
-    private final Boolean selectTargetWithNoTag;
+    private final boolean selectTargetWithNoTag;
     private final String[] filterByTagNames;
     private final Long filterByDistributionId;
 
+    private FilterParams(@NotNull Long filterByDistributionId) {
+        this(Collections.emptyList(), null, null, filterByDistributionId, null);
+    }
+
     /**
-     * Constructor.
+     * Constructor for creating filter-objects.
      *
+     * @param selectTargetWithNoTag
+     *            flag to select targets with no tag assigned
      * @param filterByInstalledOrAssignedDistributionSetId
-     *            if set, a filter is added for the given
+     *            filter by installed or assigned
      *            {@link DistributionSet#getId()}
      * @param filterByStatus
      *            if set, a filter is added for target states included by the
      *            collection
      * @param overdueState
      *            if set, a filter is added for overdued devices
      * @param filterBySearchText
-     *            if set, a filter is added for the given search text
+     *            to find targets having the text anywhere in name or
+     *            description.
      * @param selectTargetWithNoTag
```

Comment:
`selectTargetWithNoTag` param declaration is dulpicated, see line 40

Reply:
done